plugin/src/test/java/org/opensearch/ml/action/agents/TransportSearchAgentActionTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.agents;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+public class TransportSearchAgentActionTests extends OpenSearchTestCase {
+    @Mock
+    Client client;
+
+    @Mock
+    TransportService transportService;
+
+    @Mock
+    ActionFilters actionFilters;
+
+    @Mock
+    SearchRequest searchRequest;
+
+    @Mock
+    ActionListener<SearchResponse> actionListener;
+
+    TransportSearchAgentAction transportSearchAgentAction;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        transportSearchAgentAction = new TransportSearchAgentAction(transportService, actionFilters, client);
+        ThreadPool threadPool = mock(ThreadPool.class);
+        when(client.threadPool()).thenReturn(threadPool);
+        ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
+        when(threadPool.getThreadContext()).thenReturn(threadContext);
+    }
+
+    public void test_DoExecute() {
+        transportSearchAgentAction.doExecute(null, searchRequest, actionListener);
+        verify(client).search(eq(searchRequest), any());
```

Comment:
Better to verify the `client.onResponse`, also add exception test case to cover the exception part.